### PR TITLE
Punch through update

### PR DESCRIPTION
**Pull request by KevinJJackson**
### Summary
Updated mobile navigation to present the punch through effect as desktop navigation does.
Updated profile bg to match the rest of the nav.

### CODE REVIEW

File: packages/terra-consumer-nav/src/components/user-profile/UserProfile.scss [hunk 1]
```diff
@@ -58,10 +58,6 @@
     text-align: left;
     width: inherit;
 
-    @media screen and (max-width: $terra-medium-breakpoint) {
-      margin-top: 0;
-    }
-
```
Comment (alex-bezek) [inline packages/terra-consumer-nav/src/components/user-profile/UserProfile.scss]: why was this removed?

File: packages/terra-consumer-nav/src/components/user-profile/UserProfile.scss [hunk 2]
```diff
@@ -47,7 +47,7 @@
   }
 
   .popup-button {
-    background: var(--terra-consumer-nav-background, rgba(255, 255, 255, 0.5));
+    background: var(--terra-consumer-nav-background, rgba(255, 255, 255, 0.25));
```
Comment (alex-bezek) [inline packages/terra-consumer-nav/src/components/user-profile/UserProfile.scss]: I'm assuming this was correlated with some UX decision?

Comment (KevinJJackson) [reply packages/terra-consumer-nav/src/components/user-profile/UserProfile.scss]: yup, .25 is what the rest of the nav is using as the opacity. The profile link with .5 was drawing too much attention to it.

File: packages/terra-consumer-nav/src/components/user-profile/UserProfile.scss (hunk 1 above)
Comment (KevinJJackson) [reply packages/terra-consumer-nav/src/components/user-profile/UserProfile.scss]: UX decision to keep the separation between nav items and profile link via the 3px margin-top punch through

File: packages/terra-consumer-nav/src/components/nav-items/NavItem.scss [hunk 3]
```diff
@@ -19,8 +19,7 @@
     display: block;
     text-decoration: none;
 
-    &:hover,
-    &:focus {
+    &:hover {
       background: rgba(255, 255, 255, 0.4);
```
Comment (mhemesath) [inline packages/terra-consumer-nav/src/components/nav-items/NavItem.scss]: Do these need to be theme-able?

Comment (KevinJJackson) [reply packages/terra-consumer-nav/src/components/nav-items/NavItem.scss]: probably not a bad idea to be, i will update the hover effects to be theme-able
